caddyhttp/proxy/reverseproxy.go

```diff
@@ -122,10 +122,44 @@ func NewSingleHostReverseProxy(target *url.URL, without string) *ReverseProxy {
 		rp.Transport = &http.Transport{
 			Dial: socketDial(target.String()),
 		}
+	} else if keepalive != 0 {
+		rp.Transport = &http.Transport{
+			Proxy: http.ProxyFromEnvironment,
+			Dial: (&net.Dialer{
+				Timeout:   30 * time.Second,
+				KeepAlive: 30 * time.Second,
+			}).Dial,
+			TLSHandshakeTimeout:   10 * time.Second,
+			ExpectContinueTimeout: 1 * time.Second,
+		}
+		if keepalive < 0 {
+			rp.Transport.(*http.Transport).DisableKeepAlives = true
+		} else {
+			rp.Transport.(*http.Transport).MaxIdleConnsPerHost = keepalive
+		}
```

Comment:
It's up to you honestly - the reason why I kept 0 (which I agree is less intuitive), is I prefer "the default" to be a the "golang default", and if in golang 1.8, the idleconns are changed to 5, Caddy would "reflect" that. 

Another option is leaving it unset means 0, but setting `keepalive 0` actually sets it to -1 in the code.